packages/components/src/components/checkbox/checkbox.tsx

```diff
@@ -21,28 +33,21 @@ const Checkbox = React.forwardRef(
         },
         ref
     ) => {
-        const [checked, setChecked] = React.useState(defaultChecked || value);
-        const input_ref = React.useRef();
-
-        const setRef = el_input => {
-            input_ref.current = el_input;
-            if (ref) ref.current = el_input;
-        };
-
+        const [checked, setChecked] = React.useState<boolean>(defaultChecked || value);
```

Comment:
```suggestion
        const [checked, setChecked] = React.useState(defaultChecked || value);
```
it's preferable to allow TS to infer primitive types like boolean here from the value passed into useState🙂

Reply:
yes. I learned about it Today. thanks